shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1896,6 +1898,22 @@ - (void)deleteBackward {
     NSRange oldRange = ((FlutterTextRange*)oldSelectedRange).range;
     if (oldRange.location > 0) {
       NSRange newRange = NSMakeRange(oldRange.location - 1, 1);
+
+      // We should check if the last character is a part of emoji.
+      // If so, we must delete the entire emoji to prevent the text from being malformed.
+      NSRange charRange = fml::RangeForCharacterAtIndex(self.text, oldRange.location - 1);
+      UChar32 codePoint;
+      BOOL gotCodePoint = [self.text getBytes:&codePoint
+                                    maxLength:sizeof(codePoint)
+                                   usedLength:NULL
+                                     encoding:NSUTF32StringEncoding
+                                      options:kNilOptions
+                                        range:charRange
+                               remainingRange:NULL];
+      if (gotCodePoint && u_hasBinaryProperty(codePoint, UCHAR_EMOJI)) {
```

Comment:
Why is this getting applied only to emoji? 

Aren't there other multi-byte characters that are not emoji that we'd want to apply this to?

Reply:
From https://github.com/flutter/engine/pull/34508#discussion_r917184325, languages with diacritical marks will expect a backward deletion should only delete one part of the composed character, such as Thai and Hebrew.

We have two ways to distinguish this behavior from emoji's:
1) check if the byte is a diacritical mark (and other properties I don't know yet)
2) check if the byte is not in an emoji

I chose the second one to keep consistency with android (see [FlutterTextUtils.getOffsetBefore](https://github.com/flutter/engine/blob/a56d98ea78719e8ec54af8f3325740aadb953f1c/shell/platform/android/io/flutter/plugin/editing/FlutterTextUtils.java#L57-L188))

Reply:
Ahhh ok this makes sense.